Content.Shared/_RMC14/Xenonids/Parasite/SharedXenoParasiteSystem.cs

```diff
@@ -394,8 +398,15 @@ public override void Update(float frameTime)
 
             if (infected.BurstAt > time)
             {
-                // TODO RMC14 make this less effective against late-stage infections, also make this support faster incubation
-                if (infected.IncubationMultiplier < 1)
+                // Embryo dies if unrevivable when dead
+                // Kill the embryo if we've rotted or are non-humanoid
+                if (_mobState.IsDead(uid) && (!HasComp<MarineComponent>(uid) || _rotting.IsRotten(uid)))
```

Comment:
`!HasComp<MarineComponent>(uid)` This should probably check simplemob somehow instead of marine for the case of non-marine humanoids in the future like predators.